### PR TITLE
[secded_gen] Add an ecc_encode_some function

### PR DESCRIPTION
**Pull request by rswarbrick**
This is equivalent to calling ecc_encode lots of times in succession
but avoids having to pick a code and calculate bitmasks each time.

For the `rom_ctrl` usage in the second patch, using this instead of calling `ecc_encode` for each word means we run over an 8kB memory in ~4 seconds (mostly spent doing scrambling) rather than ~9 seconds.

### CODE REVIEW

**Review by msfschaffner — APPROVED** [review]
This is a good idea, let's get this in first and I'll rebase.

File: util/design/secded_gen.py [hunk 1]
```diff
@@ -392,13 +394,30 @@ def ecc_encode(codetype: str, k: int, dataword: int) -> Tuple[int, int]:
...
     log.debug(f'original hex: {hex(dataword)}')
     log.debug(f'codeword hex: {hex(int(codeword,2))}')
     return int(codeword, 2), m
 
 
+def ecc_encode_some(codetype: str,
+                    k: int,
+                    datawords: int) -> Tuple[List[int], int]:
+    m, bitmasks, invert = _ecc_pick_code(codetype, k)
+    codewords = [int(_ecc_encode(k, m, bitmasks, invert, w), 2)
```
Comment (tjaychen) [inline util/design/secded_gen.py:416]: just a nit, does this technically mean the type for `datawords` should no longer be int? But a list of ints?

Comment (rswarbrick) [reply util/design/secded_gen.py:416]: Yes it does! Oops! (A reason that I don't tend to use type annotations unless I've got some tool checking them: they're just too easy to get wrong!)

Comment (tjaychen) [reply util/design/secded_gen.py:416]: yeah completely agreed. I have a love hate relationship with the mypy checks on reggen. 
On one hand its like "gahhh! just let me merge!" but on the other hand it really helps clarify things...

**Pull request merged**